version/version.go

```diff
@@ -64,21 +49,25 @@ func parseVersionComponent(parts []string, index int, stripPrefix bool) int {
 }
 
 func VersionWithCommit(gitCommit, gitDate string) string {
-	vsn := GitTag
+	vsn := Version
 	if len(gitCommit) >= 8 {
 		vsn += "-" + gitCommit[:8]
 	}
-	if (strings.Split(GitTag, "-")[0] != "") && (gitDate != "") {
+	if (strings.Split(Version, "-")[0] != "") && (gitDate != "") {
 		vsn += "-" + gitDate
 	}
 	return vsn
 }
 
 func AsString() string {
+	// meta is not used for now, so we ignore it.
+	versionMajor, versionMinor, versionPatch, _ := parseVersion(Version)
```

Comment:
By moving the version number parsing from the init code to a lazy-evaluated place, you effectively made the final `sonicd version` command in the Makefile for checking whether the version string could be correctly parsed useless -- since it is just printing the `Version` constant.

Reply:
even if `parseVersion` is called in init, it just initializes the 3 version variables, which are only used when comparing version with the network. Both `sonicd` and `sonictool` when running version cmd, they just return the `Version`. I can change it back to have `parseVersion` in the init, but it does not affect the version cmds

Reply:
If it is in the `init` it does affect the version command in the sense that it gets executed. And if there is a parsing error in the version number that causes the parser to panic because of a very miss-shaped version tag, then we would detect it immediately at the end of the build.

The way it is now, we would only see it crashing if `AsString()` is called, which basically never happens. Would be bad if this would happen the first time in production.